Jenkinsfile

```diff
@@ -57,10 +57,29 @@ pipeline {
             }
         }
 
-       stage('Integration Tests') {
+        stage('Publish if Master') {
+            when {
+                expression {
+                    return env.BRANCH_NAME == "master"
+                }
+            }
+            environment {
+                DOCKER = credentials('rook-docker-hub')
+                AWS = credentials('rook-jenkins-aws')
+                GIT = credentials('rook-github')
+            }
+            steps {
+                sh 'docker login -u="${DOCKER_USR}" -p="${DOCKER_PSW}"'
+                sh 'build/run make -j\$(nproc) -C build/release build BRANCH_NAME=${BRANCH_NAME} GIT_API_TOKEN=${GIT_PSW}'
+                sh 'build/run make -j\$(nproc) -C build/release publish BRANCH_NAME=${BRANCH_NAME} AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW} GIT_API_TOKEN=${GIT_PSW}'
+                sh 'build/run make -j\$(nproc) -C build/release promote BRANCH_NAME=master CHANNEL=master AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW}'
```

Comment:
maybe add a comment here for why this auto promote is done for master.  quickly looking, it could appear that this logic is entirely duplicated between this new `Publish if Master` stage and the `Publish if Release` stage, but `promote` call is a key difference between them.  It was more obvious when the `promote` call was behind an inline check for master branch, but now since it's been moved to a different stage, that context is lost.

Reply:
yeah, will add a comment to keep that context